common/src/main/java/com/wynntils/screens/itemsharing/ItemSharingScreen.java

```diff
@@ -0,0 +1,260 @@
+                Models.ItemEncoding.encodeItem(wynnItem, encodingSettings);
+        if (errorOrEncodedByteBuffer.hasError()) {
+            WynntilsMod.error("Failed to encode item: " + errorOrEncodedByteBuffer.getError());
+            previewItemStack = null;
+            backgroundX = (width - Texture.ITEM_SHARING_BACKGROUND.width()) / 2;
+            clearWidgets();
+            addError(errorOrEncodedByteBuffer.getError());
+            return;
+        }
+
+        encodedItem = errorOrEncodedByteBuffer.getValue();
+
+        // Decode the item to be displayed
+        ErrorOr<WynnItem> errorOrDecodedByteBuffer = Models.ItemEncoding.decodeItem(encodedItem);
+        if (errorOrDecodedByteBuffer.hasError()) {
+            WynntilsMod.error("Failed to decode item: " + errorOrDecodedByteBuffer.getError());
+            previewItemStack = null;
+            backgroundX = (width - Texture.ITEM_SHARING_BACKGROUND.width()) / 2;
+            clearWidgets();
+            addError(errorOrDecodedByteBuffer.getError());
+            return;
+        }
+
+        WynnItem renderedItem = errorOrDecodedByteBuffer.getValue();
+
+        previewItemStack = new FakeItemStack(renderedItem, "From chat");
+
+        // Find the width of the tooltip
+        int tooltipWidth = LoreUtils.getTooltipLines(previewItemStack).stream()
+                .map(component -> McUtils.mc().font.width(component))
+                .max(Integer::compareTo)
+                .orElse(0);
+
+        // Total area the tooltip + sharing options panel + gap should cover
+        int totalRenderWidth = tooltipWidth + Texture.ITEM_SHARING_BACKGROUND.width() + 10;
+
+        // Blank space to the side of each element
+        int sideGap = Math.abs((this.width - totalRenderWidth) / 2);
+
+        tooltipX = this.width - sideGap - tooltipWidth - 20;
+        backgroundX = Math.max(0, sideGap - 10);
+
+        // Widgets need to be remade in case backgroundX has changed and the options will be misalligned
+        for (AbstractWidget widget : options) {
+            this.removeWidget(widget);
+        }
+
+        options = new ArrayList<>();
+
+        addSharingOptions();
+    }
+
+    private void shareItem(String target) {
+        if (target.equals("guild")) {
+            Handlers.Command.sendCommand("g " + encodedItem.toUtf16String());
+        } else if (target.equals("party")) {
+            Handlers.Command.sendCommand("p " + encodedItem.toUtf16String());
+        } else if (target.equals("clipboard")) {
+            McUtils.mc().keyboardHandler.setClipboard(encodedItem.toUtf16String());
+
+            McUtils.sendMessageToClient(Component.translatable("screens.wynntils.itemSharing.copied")
+                    .withStyle(ChatFormatting.GREEN));
+        } else {
+            // TODO: Save item to storage
+        }
+    }
+
+    private void addSharingOptions() {
+        // region Checkbox options
+        if (wynnItem instanceof GearItem) {
```

Comment:
This checkbox should probably be shown for charms and tomes too right? Or are they unlikely to change

Reply:
Yes. I'll patch it!